first draft of the infrastructure working group page

ChristianKuehnel opened the pull request:
A first draft of the website for the infrastructure working group

ChristianKuehnel committed:
first draft of the infrastructure working group page

ChristianKuehnel commented:
A frist draft of the website for the infrastructure working group

ChristianKuehnel committed:
review and extension

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+Initial members:
+
+* Christian Kühnel
+* ... we're recruting plese contact us at TODO!
```
Tanya should have an email alias setup in the next day or so.  We should update this with the address before we publish.

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+1. Anyone in the LLVM community (including contractors) can propose a change by 
+   creating an issue on the issue tracker. 
+1. The working group will create an statement of work (SOW) based on the  
+   proposals they want to move forward onit. The SOW will consist of a problem 
```
```suggestion
   proposals they want to move forward on. The SOW will consist of a problem 
```

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+   description and (functional and non-functional) requirements for a solution 
+   of the problem. The SOW shall be made public and discussed with the 
+   community.
+1. Once the SOW and it’s costs are clear enough, the working group will decide 
+   if and how it shall be implemented (community contribution, paid community 
+   bounty, contractor)
+    1. If it requires spending money, the working group will propose the change 
+       to the board.
+    1. The board will then review and approve the expenditure.
+    1. The Foundation will set up the contract and manage the payments.
+1. The working group will work with the contractor/community on a day-to-day 
+   basis to get the work done.
+1. The working group will approve the work and authorize payments with the 
+   treasurer (if needed).
```
The numbering here seems wrong.

Also, for any work where we expect to spend money, there will need to be an RFP process prospective contractors participate in.  We have to have a documented, fair and impartial method of determining who a paid SOW will be awarded.  I don't think the details need to be fleshed out in this document.  We need to explain here that paid work will be subject to an RFP process.  As a separate document we can explain the RFP process.  This should be done by the IWG once it has been formed.

ChristianKuehnel replied:
> The numbering here seems wrong.

What do you mean? Because it's all "1."? The renderer will [automagically replace](https://guides.github.com/features/mastering-markdown/) these with the right numbering.

> there will need to be an RFP process

Yes, this is what I understood from our discussions. I added a sentence around the RFP process and that we need to work that out.

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+  infrastructure look like?
+* Identify the need for addons for Github the community is missing right now 
+  (e.g. subscribing to labels on Github Issues).
+* Come to a decision on using Phabricator vs. GitHub Pull Requests for code
```
```suggestion
* Help the Community to come to a decision on using Phabricator vs. GitHub Pull Requests for code
```

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+  content there.
+* Create a test strategy for the LLVM project and refactor the existing build 
+  infrastructure to meet the new test strategy. Some ideas:
+  * Multi-stage CI: run “cheap” builds first, run “expensive” builds only on
```
Maybe mention pre-commit/pre-merge testing here?

sqlbyme commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,168 @@
+
+
+
+## Ideas for initial work items
```
I really like this list, thank you for putting the thought into compiling all of these points.  I wonder if this list belongs here or in another working document the IWG can work through once it is formed?

ChristianKuehnel replied:
Thx, I just put it here to collect all information and give people some ideas of what they could be working on.
Once we have some task management tool (probably GitHub Issues), I would move the list over, so we can review/discuss each option individually and also track the history/thoughts/rationale/...

ChristianKuehnel committed:
updated draft

ChristianKuehnel commented:
Hi @tlattner and @sqlbyme ,

Please review this proposal for the infrastructure working group website. I want to point to this page in the recruiting email for the working group.

Thx, Christian

ChristianKuehnel committed:
added export of original proposal

ChristianKuehnel commented:
@kwk this is a draft for the IWG website. Happy to hear your thoughts. It's mostly based on the proposal doc...

ChristianKuehnel committed:
added LNT service

ChristianKuehnel committed:
fixed path name, updated links

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+* Understanding needs of contributors and downstream users
+* IT security and privacy
+* Coordinating work in Open Source Projects
+* Tools and technologies technologies currently used in the Infrastracture 
```
"Tools and technologies ~~technologies~~ currently used in the ~~Infrastracture~~ **infrastructure**"

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+Initial members:
+
+* Christian Kühnel
+* ... we're recruting plese contact us at TODO!
```
"... we're recruting ple**a**se contact us at TODO!"

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+cancelled in any of these steps. There might also be some iterations as some 
+parts of a statement of work (SOW) are not clear or feasible.
+
+1. Anyone in the LLVM community (including contractors) can propose a change by 
```
"propose a change" - While we're at it... the "propose" part is something I'd like to define a bit more. Over the past few month, I've created a complicated, yet rather easy to use, project that allows for incremental incubator projects. That means you can take a somewhat replicated environment of our current setup and explore changes. I don't want to spoil it here. I'm planning to release a video of the work and how it can be used in the upcoming week or the following. I plan on releasing it as a video because I figure that infrastructure is complicated and people may be tempted to not read but scream more about what's bothering them. With a video they can sit back and see what's potentially possible with our infrastructure and how **they can shape the future** of the LLVM developer workflow.

ChristianKuehnel replied:
Sounds good,looking forward to your video 😄

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+This is an unsorted, unprioritized list of ideas the working group could look
+into: 
+
+* Run a survey in the community to understand the largest pain points and 
```
@ChristianKuehnel not sure but do you have something in mind that I began constructing back last year?

ChristianKuehnel replied:
Yes! I don't really have a clear picture on what the questions we should ask, but I would like to identify the major pain points of the community and get them fixed (or less painful). This would be a nice showcase for the work of the IWG and to gain trust in the community and foundation for larger topics.

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+  infrastructure look like?
+* Identify the need for addons for Github the community is missing right now 
+  (e.g. subscribing to labels on Github Issues).
+* Come to a decision on using Phabricator vs. GitHub Pull Requests for code
```
I would love to wait for the feedback on my video before ditching github PRs altogether. The possibilities of user-defined workflows is so much bigger with github than with phabricator.

ChristianKuehnel replied:
This list is just a random collection of things I've seen or people told me about. The working group needs to sieve through this and decide where we want to start. Maybe GitHub PRs might be too big as first project...

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+* Come to a decision on using Phabricator vs. GitHub Pull Requests for code
+  reviews, then implement the change.
+* Create a roadmap for the handover of the existing infrastructure to a 
+  contractor. Components ready for immediate takeover:
```
I don't like the word "takeover". It bothers me because I wonder if we lose the possibility for incremental change. That's my biggest concern. I favour to use github and incrementally implement pieces of a desirable workflow there rather than defining a workflow and handing of a large chunk to a contractor. You'll probably have a hard time changing the requirements once the takeover has happened. And I'm sure that we will change our minds at one point about a specific solution. That's just normal and has happened in every other project as well.

ChristianKuehnel replied:
We should not blindly take over existing stuff. I fully agree there. We should review and decide if it makes sense to take that over as is or shut it down or replace is with something new/better.

In all of these cases, I was in contact with the current maintainer and they would like to hand over their thing at some point in time.

Thx for pointing this out, I rephrased the section!

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+    commits where the cheap builds have passed.
+  * Set up a collector dashboard showing build results for all CI systems per 
+    commit, e.g. Treeherder or something GitHub based.
+  * Investigate if we can get a good deal from GitHub Actions with sponsored 
```
With my proposal (to be demonstrated in the video to be released) we possibly don't even need extra hardware.

ChristianKuehnel replied:
can't wait to see that 😄

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+  where we do not meet our SLOs.
+* Migrate Windows setup instructions off GnuWin32 as it’s deprecated and add
+  installation instructions users can copy-and-paste into their shell.
+* Investigate the Rust project infrastructure and see what would be helpful to 
```
Good idea. The Rust team has some kind of deal with the github folks on github actions: https://blog.rust-lang.org/inside-rust/2020/07/23/rust-ci-is-moving-to-github-actions.html

kwk commented on content/docs/infrastructure-wg.md:
```diff
@@ -0,0 +1,196 @@
+* Figure out where we need to be able to handle GDPR-related requests and
+  how that could be done.
+* Decide what to do with the mailing lists and GDPR deletion requests.
+* Decide if we want to have mandatory pre-merge testing and what it should 
```
I know that this is not the place to comment on this topic but in analogy you can probably only say that there needs to be lunch break but you cannot and should not tell what to eat.

ChristianKuehnel replied:
Not sure if I get your analogy. I guess we need to work out a tangible proposal (maybe as video 😉 ) and discuss that with the community. That proposal must explain:
* the workflows for the user
* the checks we want to run
* the expected latency of the checks
* overriding failed checks in case of some type of emergency
* ...

kwk reviewed (changes requested):
@ChristianKuehnel I have made some proposals for change and fixes some typos.

ChristianKuehnel committed:
Update content/docs/infrastructure-wg.md

Co-authored-by: Mike Edwards <mike@sqlby.me>

ChristianKuehnel committed:
fixed lint warnings

ChristianKuehnel committed:
update after reviews

ChristianKuehnel commented:
@kwk and @sqlbyme thx for the review!

I guess we can merge the first version so I can point to it from the IWG announcement email.
Happy to come back and do improvements as we go along.

ChristianKuehnel committed:
Update infrastructure-wg.md

extracted things that are changing frequently to other sites.

I created a personal Github repo to keep the stuff until we have found a permanent place to keep these...

ChristianKuehnel committed:
fixed linter warning

ChristianKuehnel committed:
upadted links

ChristianKuehnel commented:
@sqlbyme can you please merge this PR (or give me merge permissions)?
